lisa/trace.py

```diff
@@ -305,16 +280,52 @@ def has_events(self, events):
         else:
             return set(events).issubset(set(self.available_events))
 
+    def set_window(self, t_min=None, t_max=None):
+        """
+        Set trace time range to the specified values.
+
+        :param t_min: lower bound
+        :type t_min: int or float
+
+        :param t_max: upper bound
+        :type t_max: int or float
+        """
+        trace_classes = [cls for cls in self.ftrace.trace_classes if not cls.data_frame.empty]
+
+        if t_min is not None:
+            start = self._trace_end
+            for trace_class in trace_classes:
+                df = trace_class.data_frame[t_min:]
+                if not df.empty:
+                    start = min(start, df.index[0])
+            t_min = start
+        else:
+            t_min = self._trace_start
+
+        if t_max is not None:
+            end = self._trace_start
+            for trace_class in trace_classes:
+                df = trace_class.data_frame[:t_max]
+                if not df.empty:
+                    end = max(end, df.index[-1])
+            t_max = end
+        else:
+            t_max = self._trace_end
+
+        self.window = (t_min, t_max)
```

Comment:
I'm not entirely sure if we this or separate `start_time`/`end_time` members. Or if that shouldn't be named `time_window`.